src/student_registry_project/ex_1.rs

```diff
@@ -0,0 +1,384 @@
+
+            print!("Enter age: ");
+            io::stdout().flush().unwrap();
+            let mut age = String::new();
+            io::stdin()
+                .read_line(&mut age)
+                .expect("Failed to read line");
+            let age = age
+                .trim()
+                .parse::<u8>()
+                .expect("Invalid input. Please enter a number.");
+
+            print!("Enter height: ");
+            io::stdout().flush().unwrap();
+            let mut height = String::new();
+            io::stdin()
+                .read_line(&mut height)
+                .expect("Failed to read line");
+            let height = height
+                .trim()
+                .parse::<f32>()
+                .expect("Invalid input. Please enter a number.");
+
+            print!("Enter sex: 1 for male, 2 for female: ");
+            let mut sex = String::new();
+            io::stdout().flush().unwrap();
+            io::stdin()
+                .read_line(&mut sex)
+                .expect("Failed to read line");
+            let sex = sex.trim().parse::<u8>().unwrap();
+
+            if sex == 1 {
+                let student = rust_cohort.register(first_name.clone(), last_name.clone(), age, height, Sex::Male);
+                println!("student: {student:#?}");
+                io::stdout().flush().unwrap();
+                // Get course_id and enroll student
+                let course_id = get_courses(rust_cohort);
+                rust_cohort.enroll_student_in_course(student.id, course_id);
+            } else if sex == 2 {
+                let student = rust_cohort.register(first_name.clone(), last_name.clone(), age, height, Sex::Female);
+                println!("student: {student:#?}");
+                io::stdout().flush().unwrap();
+                // Get course_id and enroll student
+                let course_id = get_courses(rust_cohort);
+                rust_cohort.enroll_student_in_course(student.id, course_id);
+            } else {
+                println!("Invalid input");
+                io::stdout().flush().unwrap();
+                
+                continue;
+            }
+
+            // println!("Student {first_name} {last_name}  added successfully");
+            // io::stdout().flush().unwrap();
+
+        }
+
+        // Get student
+        if student_state == 2 {
+            let length = rust_cohort.total_students.len() as u32;
+            if length == 0 {
+                println!("No students in the cohort");
+                io::stdout().flush().unwrap();
+
+            } else {
+                let len = length - 1;
+                print!("Enter student id(Range 0 to {len}): ");
+                io::stdout().flush().unwrap();
+
+                let mut id = String::new();
+                io::stdin().read_line(&mut id).expect("Failed to read line");
+                let id = id.trim().parse::<u32>().unwrap();
+                let student = rust_cohort.get_student_by_id(id);
```

Comment:
In critical mission applications, Consider handling cases where value could be Some or None. This function returns an Option<Student>